sentinel-dashboard/src/main/java/com/alibaba/csp/sentinel/dashboard/controller/v2/ParamFlowRuleControllerV2.java

```diff
@@ -0,0 +1,271 @@
+                .orElse(true);
+            // If error occurred or cannot retrieve machine info, return true.
+        } catch (Exception ex) {
+            return true;
+        }
+    }
+
+    @GetMapping("/rules")
+    public Result<List<ParamFlowRuleEntity>> apiQueryAllRulesForMachine(HttpServletRequest request,
+                                                                        @RequestParam String app,
+                                                                        @RequestParam String ip,
+                                                                        @RequestParam Integer port) {
+        AuthUser authUser = authService.getAuthUser(request);
+        authUser.authTarget(app, PrivilegeType.READ_RULE);
+        if (StringUtil.isEmpty(app)) {
+            return Result.ofFail(-1, "app cannot be null or empty");
+        }
+        if (StringUtil.isEmpty(ip)) {
+            return Result.ofFail(-1, "ip cannot be null or empty");
+        }
+        if (port == null || port <= 0) {
+            return Result.ofFail(-1, "Invalid parameter: port");
+        }
+        if (!checkIfSupported(app, ip, port)) {
+            return unsupportedVersion();
+        }
+        try {
+            return Result.ofSuccess(ruleProvider.getRules(app));
+        } catch (ExecutionException ex) {
+            logger.error("Error when querying parameter flow rules", ex.getCause());
+            if (isNotSupported(ex.getCause())) {
+                return unsupportedVersion();
+            } else {
+                return Result.ofThrowable(-1, ex.getCause());
+            }
+        } catch (Throwable throwable) {
+            logger.error("Error when querying parameter flow rules", throwable);
+            return Result.ofThrowable(-1, throwable);
+        }
+    }
+
+    private boolean isNotSupported(Throwable ex) {
+        return ex instanceof CommandNotFoundException;
+    }
+
+    @PostMapping("/rule")
+    public Result<ParamFlowRuleEntity> apiAddParamFlowRule(HttpServletRequest request,
+                                                           @RequestBody ParamFlowRuleEntity entity) {
+        AuthUser authUser = authService.getAuthUser(request);
+        authUser.authTarget(entity.getApp(), PrivilegeType.WRITE_RULE);
+        Result<ParamFlowRuleEntity> checkResult = checkEntityInternal(entity);
+        if (checkResult != null) {
+            return checkResult;
+        }
+        if (!checkIfSupported(entity.getApp(), entity.getIp(), entity.getPort())) {
+            return unsupportedVersion();
+        }
+        entity.setId(null);
+        entity.getRule().setResource(entity.getResource().trim());
+        Date date = new Date();
+        entity.setGmtCreate(date);
+        entity.setGmtModified(date);
+        try {
+            entity = repository.save(entity);
+            publishRules(entity.getApp());
+            return Result.ofSuccess(entity);
+        } catch (ExecutionException ex) {
```

Comment:
Just catch `Exception` (or `Throwable`) here is enough. You may refer to `FlowControllerV2`.